From: Ben Hutchings <bhutchings@solarflare.com>
Date: Thu, 13 Sep 2012 01:11:24 +0100
Subject: [PATCH v2 047/244] sfc: Use efx_mcdi_mon() to find efx_mcdi_mon
 structure from efx_nic
Patch-mainline: v3.12-rc1
Git-commit: e847b53e9e3c6f7818fe2fcd1082148423020ccc
References: bsc#909618 FATE#317521

This needs to be done before we separate MCDI from siena_nic_data.

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/sfc/mcdi_mon.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/net/ethernet/sfc/mcdi_mon.c
+++ b/drivers/net/ethernet/sfc/mcdi_mon.c
@@ -400,8 +400,7 @@ fail:
 
 void efx_mcdi_mon_remove(struct efx_nic *efx)
 {
-	struct siena_nic_data *nic_data = efx->nic_data;
-	struct efx_mcdi_mon *hwmon = &nic_data->hwmon;
+	struct efx_mcdi_mon *hwmon = efx_mcdi_mon(efx);
 	unsigned int i;
 
 	for (i = 0; i < hwmon->n_attrs; i++)
